persistence/src/main/java/edu/unc/lib/dl/persist/services/destroy/DestroyObjectsJob.java

```diff
@@ -84,10 +112,18 @@ public void run() {
             // convert each destroyed obj to a tombstone
             for (PID pid : objsToDestroy) {
                 RepositoryObject repoObj = repoObjLoader.getRepositoryObject(pid);
+
+                assertCanDestroy(agent, repoObj, aclService);
```

Comment:
Don't delete the binaries until the fedora cleanup has been successful, otherwise the rollback can result in objects with missing files.